shared/src/commonMain/kotlin/org/hyperskill/app/profile/domain/repository/CurrentProfileStateRepository.kt

```diff
@@ -0,0 +1,14 @@
+package org.hyperskill.app.profile.domain.repository
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import org.hyperskill.app.core.domain.repository.StateRepository
+import org.hyperskill.app.profile.domain.model.Profile
+
+interface CurrentProfileStateRepository : StateRepository<Profile>
+
+fun CurrentProfileStateRepository.observeHypercoinsBalance(): Flow<Int> =
+    changes
+        .distinctUntilChanged()
+        .map { it.gamification.hypercoinsBalance }
```

Comment:
I think it's better to change order of `distinctUntilChanged` and `map` methods to emit changes only when `hypercoinsBalance` is changed, in current implementation it will be emitted for example, when `trackId` is changed. Or I don't understand `distinctUntilChanged` method?

Reply:
@vladkash 
Yes, you are right, the value will be produced when any of the class fields are changed.
It's better to change the order.